### PR TITLE
Add GridMask augmentation

### PR DESCRIPTION
**Pull request by haqishen**
An implementation of GridMask : https://arxiv.org/abs/2001.04086

demo： https://www.kaggle.com/haqishen/gridmask

### CODE REVIEW

**Comment by Dipet** [comment]
Please add information about transform like described [here](https://github.com/albumentations-team/albumentations#contributing)
And reformat code using black, for this purposes you can use [pre-commit hook](https://github.com/albumentations-team/albumentations/blob/master/docs/contributing.rst#to-create-a-pull-request)

File: albumentations/augmentations/transforms.py [hunk 1]
```diff
@@ -1497,6 +1498,98 @@ def get_transform_init_args_names(self):
...
+        self.rotate = rotate
+        self.mode = mode
+        self.masks = None
+        self.rand_h_max = []
+        self.rand_w_max = []
+
+    def init_masks(self, height, width):
+        if self.masks is None:
+            self.masks = []
+            for n, n_g in enumerate(range(self.num_grid[0], self.num_grid[1] + 1, 1)):
```
Comment (Dipet) [inline albumentations/augmentations/transforms.py]: When images have different shapes this type of caching values is wrong

File: albumentations/augmentations/transforms.py [hunk 2]
```diff
@@ -1497,6 +1498,98 @@ def get_transform_init_args_names(self):
...
+    |  https://arxiv.org/abs/2001.04086
+    |  https://github.com/akuxcw/GridMask
+    """
+
+    def __init__(self, num_grid=3, fill_value=0, rotate=0, mode=0, always_apply=False, p=0.5):
+        super(GridMask, self).__init__(always_apply, p)
+        if isinstance(num_grid, int):
+            num_grid = (num_grid, num_grid)
+        if isinstance(rotate, int):
+            rotate = (-rotate, rotate)
```
Comment (Dipet) [inline albumentations/augmentations/transforms.py]: You can use here our `to_tuple` method

**Comment by Dipet** [comment]
Please look to #520 This PR implements the same things. I think you need to decide with author whats to do with this transformation.

**Comment by tatigabru** [comment]
Hi @haqishen, yes, in #520 PR implemented the same things, but I added more user-friendly parameters. And I did not add rotation to avoid overloading it, as there are other augmentations that cover it (you can rotate the image and then apply grid mask, with the same effect)
If rotation is needed within this augmentation, maybe when I finish my PR you could add your rotation to it, for example? I do not know, what to do for the case of the double-PR

**Comment by Dipet** [comment]
I think is better wait when we will merge #520, then @haqishen can add his improvements like different transform mods and rotations.

**Comment by haqishen** [comment]
I read the code of #520, I think it's better to add my code on #520 as well
Will do it later.
I'm closing this PR.

**Pull request closed without merging**